packages/react-core-auth/src/components/Authenticator/Machine/MachineContext.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useCallback } from 'react';
+import { useInterpret, useSelector } from '@xstate/react';
+import {
+  AuthenticatorMachineOptions,
+  AuthMachineState,
+  getNextServiceFacade,
+  createAuthenticatorMachine,
+} from '@aws-amplify/ui';
+
+import { MachineContextType, UseMachineSelector, UseMachine } from './types';
+import { defaultComparator, getComparator } from './utils';
+
+export const USE_MACHINE_ERROR =
+  '`useMachine` must be used inside an `Authenticator.Provider`.';
+
+const MachineContext = React.createContext<MachineContextType | null>(null);
+
+export function useMachine(selector?: UseMachineSelector): UseMachine {
+  const context = React.useContext(MachineContext);
+
+  if (!context) {
+    throw new Error(USE_MACHINE_ERROR);
+  }
+
+  const { service } = context;
+  const { send } = service;
+
+  const xstateSelector = useCallback(
+    (state: AuthMachineState) => getNextServiceFacade({ send, state }),
+    [send]
+  );
+
+  const comparator = selector ? getComparator(selector) : defaultComparator;
+  const facade = useSelector(service, xstateSelector, comparator);
+
+  return facade;
```

Comment:
Now you're just showing off ;)

Reply:
Tbh we might be able to just return the `context` now since:
- 1) way less interactions are happening with the machine 
- 2) the machine data used is mostly javascript primitives that wouldn't cause re-renders unless they changed irregardless of a selector